### PR TITLE
[WIP] gs: fix import-url for directories

### LINKED ISSUE #7638: import-url: importing directory from GCS bucket downloads to parent directory
# Bug Report

## Description
This could be a user error but something is confusing (and seems wrong) with `dvc import-url gs://my-bucket/my-external-data-dir/` and where the files get downloaded to. Checksums are computed correct but the downloaded directory is empty (`WARNING: 'my-external-data-dir' is empty.`).

### Reproduce
Create a new git repo
```sh
mkdir my_repo
cd my_repo
git init
touch README.md
git add README.md
git commit -m "Add README.md"
```

Initialize dvc inside git repo
```sh
python -m venv .venv && source .venv/bin/activate && pip install dvc[gs]
dvc init
```

Create a Google Storage bucket and add some files to a directory.
```sh
gsutil mb gs://my-tmp-bucket-for-dvc-bug
touch file.txt
gsutil cp file.txt gs://my-tmp-bucket-for-dvc-bug/my-external-data-dir/file.txt
rm file.txt
```

Import external data into DVC workspace under data/
```sh
mkdir data && cd data
dvc import-url gs://my-tmp-bucket-for-dvc-bug/my-external-data-dir
```

File directory got downloaded, but with wrong path. It's outside the git repo.
```
ls ../../my-tmp-bucket-for-dvc-bug/my-external-data-dir/
```

### Expected
my-external-data-dir.dvc and my-external-data-dir/ with downloaded files within it, and the directory added to .gitignore, under data/

### Actual
```sh
dvc import-url gs://my-tmp-bucket-for-dvc-bug/my-external-data-dir
Importing 'gs://my-tmp-bucket-for-dvc-bug/my-external-data-dir' -> 'my-external-data-dir'
WARNING: 'my-external-data-dir' is empty.
```

```sh
cat my-external-data-dir.dvc 
md5: 7fa09692529be14144a721c582399a3e
frozen: true
deps:
- md5: 081ca9f4767585c8dfe3c9baa0d41f7a.dir
  size: 0
  nfiles: 1
  path: gs://my-tmp-bucket-for-dvc-bug/my-external-data-dir
outs:
- md5: d751713988987e9331980363e24189ce.dir
  size: 0
  nfiles: 0
  path: my-external-data-dir
```

### Environment information
**Output of `dvc doctor`:**

```console
DVC version: 2.10.1 (pip)
---------------------------------
Platform: Python 3.9.7 on Linux-5.13.0-40-generic-x86_64-with-glibc2.34
Supports:
	gs (gcsfs = 2022.3.0),
	webhdfs (fsspec = 2022.3.0),
	http (aiohttp = 3.8.1, aiohttp-retry = 2.4.6),
	https (aiohttp = 3.8.1, aiohttp-retry = 2.4.6)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/mapper/vgubuntu-root
Caches: local
Remotes: None
Workspace directory: ext4 on /dev/mapper/vgubuntu-root
Repo: dvc, git
```

### PR DESCRIPTION
**Pull request by dtrifiro**
fixes #7638

### CODE REVIEW

**Review by daavoo — COMMENTED** [review]
We should be catching this external data stuff in the cloud tests (https://github.com/iterative/dvc-gs). 
Seems that we are not really testing external data at all?
Similar thing happened in https://github.com/iterative/dvc/pull/7563

**Comment by efiop** [comment]
@daavoo Indeed, we specifically skip import_dir test in https://github.com/iterative/dvc-gs/blob/564592e28be38c8d45f7c04f0b07d0065a65f1b2/dvc_gs/tests/test_dvc.py#L39 I think this is for legacy reasons from way back when we used etags for dir hashes. We now use regular md5s, so might be able to make it work for gs.

**Comment by efiop** [comment]
@dtrifiro The testing for gs is a bit complicated, because we are in the middle of creating plugins, so the tests run in https://github.com/iterative/dvc-gs (they have the creds and stuff there). If you don't happen to have set it up before, you could create a test PR in dvc-gs and make it install dvc from your fork and run the tests there, just to confirm that it works before merging this.

**Comment by dtrifiro** [comment]
Seems  there are some failures https://github.com/iterative/dvc-gs/runs/6209906045?check_suite_focus=true. Will mark this PR as draft and investigate later

**Comment by dberenbaum** [comment]
> We should be catching this external data stuff in the cloud tests (https://github.com/iterative/dvc-gs).
> Seems that we are not really testing external data at all?
> Similar thing happened in #7563

Should we create a separate issue for this?

**Comment by daavoo** [comment]
This needs to be moved to https://github.com/iterative/dvc-objects

File: dvc/fs/__init__.py [hunk 1]
```diff
@@ -124,6 +124,9 @@ def get_cloud_fs(repo, **kwargs):
         # For WebDAVFileSystem, provided url is the base path itself, so it
         # should be treated as being a root path.
         fs_path = cls.root_marker
+    if issubclass(cls, GSFileSystem):
+        # For GSFileSystem there's no need to strip
+        fs_path = url
```
Comment (efiop) [inline dvc/fs/__init__.py:129]: Hm, could you explain why we are not modifying _stip_protocol instead?

Comment (dtrifiro) [reply dvc/fs/__init__.py]: This was initially added due to a failing API-related test. It's no longer necessary, so only the https://github.com/iterative/dvc-objects/pulls/41 is required

**Comment by dtrifiro** [comment]
Closing this as this was merged in iterative/dvc-objects#41. Just need to wait for `dvc-data` to be bumped

**Pull request closed without merging**